app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/Indexer/ModifySelectInProductPriceIndexFilter.php

```diff
@@ -104,8 +105,16 @@ public function aroundModifyPrice(
                     ['inventory_stock' => $stockTable],
                     'inventory_stock.sku = product_entity.sku',
                     []
+                )->joinLeft(
+                    ['legacy_stock_item' => $this->resourceConnection->getTableName('cataloginventory_stock_item')],
+                    'product_entity.entity_id = legacy_stock_item.product_id',
+                    ['use_config_manage_stock', 'manage_stock']
                 );
-                $select->where('inventory_stock.is_salable = 0 OR inventory_stock.is_salable IS NULL');
+                $stockSettingsCondition = '(legacy_stock_item.use_config_manage_stock = 0 AND legacy_stock_item.manage_stock = 1)';
+                if (1 === $globalManageStock) {
+                    $stockSettingsCondition .= ' OR legacy_stock_item.use_config_manage_stock = 1';
+                }
```

Comment:
You can use here
\Magento\InventoryExportStock\Model\ResourceModel\ManageStockCondition